### ISSUE TITLE
[BUG] device testing using playwright 

### ISSUE THREAD

**Issue by dimkin-eu** [microsoft/playwright#9216]
**Context:**
- Playwright Version: 1.15
- Operating System: macos Catalina
- Node.js version: v14.16.0
- Browser: Version 15.0 (15612.1.29.41.4, 15612)

**Code Snippet**

I was trying to understand - how reliable is testing mobile versions using PW
so, I've opened https://habr.com/ru/news/ in
- Xcode simulator iPhone 8 plus ios 14.4
- Safari -> iPhone 8Plus - ios14 emulator
- playwright device ` use: devices["iPhone 8 Plus"],`

which produced such results

<img width="1343" alt="Screenshot 2021-09-29 at 20 42 33" src="https://user-images.githubusercontent.com/7899764/135328712-eb328845-01c6-4c1a-be37-82a880d69c4f.png">
<img width="1339" alt="Screenshot 2021-09-29 at 20 42 03" src="https://user-images.githubusercontent.com/7899764/135328740-b1fb490a-a2bf-431b-8df2-23b833e31e9e.png">

As you can see - they are pretty different
also, I've noticed, that [deviceDescriptorsSource](https://github.com/mxschmitt/playwright/blob/master/src/server/deviceDescriptorsSource.json) have many inaccurate and outdated UAs like for iPhone X
```
  "iPhone X": {
    "userAgent": "Mozilla/5.0 (iPhone; CPU iPhone OS 11_0 like Mac OS X) AppleWebKit/604.1.38 (KHTML, like Gecko) Version/15.0 Mobile/15A372 Safari/604.1",
```

which initial OS was 11.1 ( not 11.0, as in UA ) and 15A372 is iPhone 6,2
So if some webpage has some tweaks, based on UA, they will render something different

**Describe the bug**
- minor bug - outdated user-agents
- major one - rendered in playwright differs from a device (rendered part is way larger, that in the real device ( not tested yet ios15, which will bring more fun)) and probably could match emulator in the browser, which is not a big helper

**Comment by pavelfeldman** [comment]
I can repro this behavior with `node lib/cli/cli wk habr.com/ru/hubs --device="iPhone 8"` on ToT, but strangely not with 1.15.0. Customers are seeing it on 1.15.0 though.

**Comment by dimkin-eu** [comment]
npx playwright wk https://playwright.dev/ --device="iPhone 8 Plus"
<img width="905" alt="Screenshot 2021-09-30 at 19 36 52" src="https://user-images.githubusercontent.com/7899764/135495846-c606dd36-7129-4e00-ba86-62d1d8af12e6.png">

**Comment by aslushnikov** [comment]
* Fixed on tip of tree with WebKit roll to 1550 - https://github.com/microsoft/playwright/commit/c63348bd0389ab6dda933dca3af3f679da0f7252
* fixed on the release-1.15 branch as well - https://github.com/microsoft/playwright/commit/83cfc2fd633d8e1e239e94cdd41e44b714c46dee
